runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -758,5 +775,30 @@ public void updateLatestCheckpointIfLater(UUID checkPointId) {
             }
         }
     }
+
+    /**
+     * This queue implementation is to be used by single threaded exeuctors
+     * to restrict the amount of pending job submissions.
+     */
+    public class BoundedQueue<E> extends ArrayBlockingQueue<E> {
+
+        public BoundedQueue(int size) {
+            // This queue will be used to processes a consecutive range
+            // of elements: FIFO order is needed. Thus, fair=true
+            super(size, true);
```

Comment:
I think fair's FIFO relates to threads access order not queue's elements. Doesn't this data structure specifically mentioned to be used single threaded?

Reply:
A queue ADT is by definition FIFO. I though there were multiple producers, but I realized that its a single producer, single consumer, so it really doesn't matter.

I'm thinking about removing the executor, I don't feel like buffering is really necessary here. What do you think ?

Reply:
@Maithem I think there is a little extra overhead when fair is true but shouldn't matter much considering it is used in single threads. 
As for removing the executor, keeping it around adds the resilience of keeping the pool size to exactly one without extra effort in case of a necromancer thread terminating. A mild positive point for keeping it. I'm guessing you're concerned with executor's overhead that may not add much value?